File: 824._Goat_Latin.py
class Solution:
    def check(self,word):
        if word[0] in 'aeiouAEIOU':
            return word+"ma"
        else:
            return word[1:]+word[0]+"ma"
    def toGoatLatin(self, S):
        list_s=list(S.split(" "))
        for i in range(len(list_s)):
            x=self.check(list_s[i])
            r='a'*(i+1)
            x=x+r
            list_s[i]=x
        return ' '.join(list_s)
        

File: 824._Goat_Latin.cpp
class Solution {
public:
    string toGoatLatin(string S) {
        istringstream in(S);
        string ans, word, app = "ma", table = "AEIOUaeiou";
        while (in >> word) 
        {
            app += 'a';
            
            if (table.find_first_of(word[0]) != -1) 
                ans += word + app + ' ';
            else 
                ans += (word.size() == 1 ? word : word.substr(1, word.size() - 1) + word[0]) + app + ' ';
        }
        ans.pop_back();
        return ans;
    }
};